x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizerTests.java

```diff
@@ -1611,6 +1612,22 @@ public void testMatchOptionsPushDown() {
         assertThat(actualLuceneQuery.toString(), is(expectedLuceneQuery.toString()));
     }
 
+    public void testQStrOptionsPushDown() {
+        String query = """
+            from test
+            | where QSTR("first_name: Anna", {"lenient": "false", "fuzzy_max_expansions": 4, "minimum_should_match": 3})
```

Comment:
It would be good to check that _all_ possible options get correctly translated to the `QueryBuilder`, so we don't have to explicitly test them in other places.

Adding all possible options with some value, and then checking that the passed value gets to the **QueryBuilder** should be enough here.